### DISCUSSION TITLE
Looking for contributors and the road ahead

### DISCUSSION THREAD

**Discussion by rolandjitsu** [react-dropzone/react-dropzone#1136] [Announcements]
New year, new goals!

This announcement is just a port of https://github.com/react-dropzone/react-dropzone/issues/479 to discussions and also a chance to do a little update, so please read on.

### Feb 2022
The previous message still holds true. This lib is still being actively maintained, even though it doesn't seem like there's much activity at times.

The plan for this year is to:
* [x] Add support for [React 18](https://reactjs.org/blog/2021/06/08/the-plan-for-react-18.html)
* [x] Update the dev deps to latest for all repos (things move fast, so there were quite a few changes)
* [x] Fix styleguide issues in all repos
* [x] Use [File System Access API](https://developer.mozilla.org/en-US/docs/Web/API/File_System_Access_API) to open the file picker instead of triggering click on an `<input type="file">` (note that this is a draft, so only supported in some browsers)
* [x] Standardise how the `accept` attr works ([File System Access API](https://developer.mozilla.org/en-US/docs/Web/API/File_System_Access_API) doesn't support file extensions, but the `<input>` does)
* [x] Address the [issues](https://github.com/react-dropzone/react-dropzone/issues) which seem to be causing the biggest headaches
* [ ] Improve code readability
* [x] Revamp the docs for [react-dropzone](https://github.com/react-dropzone/react-dropzone)
* [x] Revamp the logo?
* [x] Add docs site for [file-selector](https://github.com/react-dropzone/file-selector) and [attr-accept](https://github.com/react-dropzone/attr-accept)
* [ ] Add e2e tests and run against saucelabs/browserstack 
* [x] Improve community health (add issue/PR templates, code of conduct, funding, etc)
* [x] Improve organisation page
* [ ] Work on the OpenCollective plans and try to get more backers/sponsors
* [ ] Find at least 1 collaborator that can help with PR reviews and other chores
* [ ] Others (community suggestions, etc)

With that said, I'm still looking for contributors that can help with PR reviews and chores every now and then. So if you're interested, please email me or comment in this thread.

### Jan 2019
This issue/message is by no means reason for concern, the lib is still being maintained. The issue/message is here to encourage more ppl. to contribute and help with arising issues.

There is no need to add more comments to this topic, instead, if you feel like something needs to be fixed or changed, feel free to make a PR.

**Comment by rolandjitsu** [comment]
Looking back at this, I realise that we/I managed to achieve pretty much nothing that's mentioned above 😞 

At this point, the priority should probably be to address the major bugs that are reported and merge/close the PRs that are currently open.

**Reply by rolandjitsu** [reply]
Immediate actions to take are:
1. Review PRs
2. Fix CI pipeline failures
3. https://github.com/react-dropzone/react-dropzone/discussions/1186 - make use of file system API disabled by default
4. Triage the rest of the reported bugs

**Reply by rolandjitsu** [reply]
Current progress on the above:
1. [x] Review PRs
2. [x] Fix CI pipeline failures
3. [x] https://github.com/react-dropzone/react-dropzone/discussions/1186 - make use of file system API disabled by default
4. [ ] Triage the rest of the reported bugs

**Reply by rolandjitsu** [reply]
Managed to triage the rest of the reported bugs as well. The next steps are:
1. Bring ESM to all repos (@jonkoops has helped on file-selector; I'm trying to do it for react-dropzone while keeping support for commonjs, but may end up dropping that)
2. Review the issues again and tackle one by one
3. Finally move the docs for react-dropzone to some other platform/framework (I hope that I'll finally get some time for this or someone else will tackle it)

**Reply by jonkoops** [reply]
I'll see if I can make some time next weekend to also get the other modules ported over to ESM, plus the other enhancements that I've done for `file-selector`. Perhaps we should focus on clearing out all open issues on `file-selector` first, get a 2.0 release out and then look at the rest. WDYT @rolandjitsu?

**Reply by rolandjitsu** [reply]
I can also help with that, if I have time as well 😆 

I think `file-selector`'s issues are in a good state (none left!). I'm working through `react-dropzone`'s issues one by one, whichever seems easiest to reproduce and fix.

Btw, `file-selector` will probably be 3.0 as there was another change that was "breaking" (the change could have been a fix, but it may break some ppl's usage).
